File: py/clientAis/games/GameManager.py
from typing import Union

from clientAis.communications.GameState import GameState

class GameManager:
    def __init__(self):
        self.gameState: Union[GameState, None] = None
        self.currentPlayerId: Union[str, None] = None

    def setNextCompleteGameState(self, gameState: GameState):
        self.gameState = gameState

    def setCurrentPlayerId(self, playerId: str):
        self.currentPlayerId = playerId

File: jva/src/clientAis/games/GameManager.java
package clientAis.games;

import clientAis.communications.game_data.GameState;

import java.util.Optional;

public class GameManager {

    public Optional<String> currentPlayerId;
    public Optional<GameState> gameState;

    public GameManager() {
        this.currentPlayerId = Optional.empty();
        this.gameState = Optional.empty();
    }

    public void setCurrentPlayerId(String currentPlayerId) {
        this.currentPlayerId = Optional.of(currentPlayerId);
    }

    public void setNextCompleteGameState(GameState gameState) {
        this.gameState = Optional.of(gameState);
    }
}
